src/terminal.ts

```diff
@@ -81,10 +81,19 @@ export class FlatpakTaskTerminal implements vscode.Pseudoterminal {
 
   spawn(command: Command): void {
     this.writeEmitter.fire(`${command.toString()}`)
-    const proc = child_process.spawn(command.name, command.arguments, {
-      cwd: command.cwd,
-      shell: '/usr/bin/bash',
-    })
+    let proc
+    if (command.isSandboxed) {
+      proc = child_process.spawn("flatpak-spawn", ["--host", command.name, ...command.arguments], {
+        cwd: command.cwd,
+        shell: '/usr/bin/bash',
+      })
+    }
+    else {
+      proc = child_process.spawn(command.name, command.arguments, {
+        cwd: command.cwd,
+        shell: '/usr/bin/bash',
+      })
```

Comment:
Agh, probably needs fixing regarding the changes i have made in https://github.com/bilelmoussaoui/flatpak-vscode/pull/16/files#diff-80b106dfebcf0f2ffdc8f0b21bc9faa0dea9f9b051480bc437170a7ebf9160afR24-R30 and i think that's the proper place to handle it